[Team-20 BE] Airbnb 프로젝트 1주차(수요일) 리뷰 요청

ikjo93 opened the pull request:
안녕하세요 Dan!! 이번 Airbnb 프로젝트 과정 20팀에서 백엔드 역할을 하고있는 익조입니다!! 😁 이번에 처음 뵙게 되어 무척 반가웠고 점심 시간도 내주시어 프로젝트 관련 좋은 말씀도 주셔서 감사드렸습니다!! ✨

이번 프로젝트와 관련하여 월요일~화요일까지 진행한 내용들을 README 파일에 작성했습니다. 우선적으로 배포를 위한 AWS 인프라 구성을 위주로 작업을 했었던 터라 애플리케이션 관련 로직이 거의 없는 점 양해부탁드립니다. 😂 확인해주시고 이와 관련하여 조언이나 더 나은 방향성(키워드 등)을 주시면 감사드리겠습니다. :)

앞으로 3주간 잘 부탁드리겠습니다 Dan!!

mjsdo committed:
init: 프로젝트 초기화

ikjo93 committed:
init : Spring boot 프로젝트 초기화

ikjo93 committed:
chore : application 파일 업데이트

ikjo93 committed:
feat : 도메인 모델링 및 테이블 설계 작업(미완성)

* 연관관계
 - Member : Reservation = 1 : N
 - Reservation : Accommodation = 1 : 1
 - Accommodation : Facility = 1 : 1

* TODO
 - 숙소(Accommodation)별 체크인 및 체크아웃 시간 관리 전략 수립

ikjo93 committed:
fix : 오탈자 수정

ikjo93 committed:
docs : 1주차 수요일 PR 관련 진행사항 작성

Hyune-c commented on be/README.md:
````diff
@@ -0,0 +1,25 @@
+> 숙소(Accommodation): 숙소 관련 정보(가격, 위치, 이미지 파일경로 등)를 가집니다. 숙소 관련 시설 정보(Facility)와 일대일 관계입니다.
+> 시설(Facility): 숙소 관련 시설에 대한 정보(욕실의 개수, 인터넷 사용 가능유무 등)를 가집니다.
+> ```
+> ![image](https://user-images.githubusercontent.com/82401504/170213984-0840cf7b-9cdd-4f20-8687-2c3065dde66b.png)
````
`nit`

- `reservation_date_time` 이라는 단어에 생소한 느낌이 들었는데, 생각해보니까 저는 `datetime`은 그냥 `at`으로 썻어서 그런거 같아요.
    - 예를 들면 `update_at`과 같이 사용했습니다.
    - 저와 같이 할 필요는 없지만 `date_time`이라고 구분하는 것 보다는 `datetime` 이 더 나은 느낌이에요.
- `positionx` 는 가독성이 떨어집니다. 공백이 필요해보입니다.

ikjo93 replied:
넵!! date_time 보다 datetime 또는 at이 적합해보이고 posittionx 역시 position_x로 함이 적합하다고 생각됩니다!! (사실 하이버네이트 자동 ddl 기능으로 생성된 필드라.. 아직 수정을 하지 못했습니다.. 😂)

Hyune-c commented on be/src/main/java/codesquad/airbnb/domain/Accommodation.java:
```diff
@@ -0,0 +1,39 @@
+@Getter
+public class Accommodation {
+
+    @Id @GeneratedValue
```
`@GeneratedValue`의 strategy를 명시하는 것과 하지 않는 것에는 무슨 차이가 있을까요?

Hyune-c replied:
https://hyune.gitbook.io/hyune-wiki/and/codesquad/22reviewer/airbnb

네. 이론을 어느정도 공부하신 것 같네요.
링크의 `JPA entity의 식별자 기본 생성 전략에 대한 리뷰` 를 참고해주세요.

ikjo93 replied:
감사합니다!! 사실 기본 값 AUTO로 설정 시 자동으로 IDENTITY 전략을 취할 것으로 알았는데(넘겨짚어) TABLE 전략을 취하고 있었군요...😂

Hyune-c commented on be/src/main/java/codesquad/airbnb/domain/Accommodation.java:
```diff
@@ -0,0 +1,39 @@
+    private String imagePath;
+
+    @Column(name = "price_day")
+    private int pricePerDay;
```
굳이 이름을 나눈 이유가 무엇일까요? 🤔

ikjo93 replied:
개인적으로 테이블 필드가 priceperday라고 하면 부자연스러울 것 같아서 나누었는데... 처음 price_per_day로 하려다가 좀 부자연스러운 느낌이라 price_day로 바꾸었는데 마찬가지로 부자연스러운 느낌같습니다.. 오히려 priceperday 그냥 그대로 사용하는 것이 보이는 것 같습니다.

Hyune-c commented on be/src/main/java/codesquad/airbnb/domain/Accommodation.java:
```diff
@@ -0,0 +1,39 @@
+    @Column(name = "price_day")
+    private int pricePerDay;
+    private int positionX;
+    private int positionY;
```
entity의 필드를 primitive 형으로 사용한 이유가 무엇일까요? 🤔

ikjo93 replied:
Integer : null로 초기화할 수 있음 그러나 boxing, unboxing 처리 필요, int 대비 메모리 높음
int : null로 초기화할 수 없음 그러나 별도 boxing, unboxing 처리 필요 없음 Integer 대비 메모리 적음

가격과 숙소 좌표값이 null로 초기화될 일이 있을까?(좁은 견해이지만) 라는 의문이 들어 아주 작은 효과지만 조금 더 성능이 나은 int를 사용하고자 했습니다. 좁은 견해이기 때문에 만약을 대비해서(null로 초기화 될 수 있는) Integer로 쓸수도 있겠지만 학습단계이기 때문에 나름 소신(?)껏 int를 사용해보았는데 제가 생각하지 못했었던 부분이 있으면 말씀주시면 감사드리겠습니다. 🙏

아 그리고 아까 얘기(startagey 생략)가 나온 김에 변명은 아니지만..😂 굳이(?) 말씀드리면 영한님 JPA 강의에서도 가격이나 재고수량 등의 정보는 int로 사용하시는 것을 참고하였습니다.(그렇다고 이 말이 옳다고 주장하는 것은 아닙니다.)

Hyune-c replied:
https://hyune.gitbook.io/hyune-wiki/and/codesquad/22reviewer/airbnb

사실 이 리뷰는 wrapper 형을 유도하기 위해 달은 것이긴 한데, 조금 더 조사해보니까 코드 스타일 차이로 가도 될 것 같습니다.
링크의 `Entity의 field type은 primive여야 될까, wrapper여야 될까에 대한 리뷰`를 참고해주세요.

Hyune-c commented on be/src/main/java/codesquad/airbnb/domain/Facility.java:
```diff
@@ -0,0 +1,36 @@
+    private boolean kitchenFlag;
+    private boolean internetFlag;
+    private boolean airConditionerFlag;
+    private boolean hairDryerFlag;
```
`nit`
혹시 airConditioner나 hariDryer는 한 단어로 간주하는건 어떨까요?
저는 username 같이 단어는 2개지만 하나의 의미를 가진 경우는 합치곤 합니다.

ikjo93 replied:
넵! 말씀주신 내용이 적합하다고 생각합니다!! 아울러 프론트엔드 분들과 요구사항 분석 관련 논의를 하면서도 이 내용들은 하나의 문자열로 표기하는 것이 적합하다는 얘기가 나왔습니다!

Hyune-c reviewed (changes requested):
수고하셨습니다.

jpa 에 관련된 테스트를 많이 해보시면 좋을 것 같습니다.

ikjo93 commented:
미처 생각하지 못했었던 부분들을 짚어주셔서 감사드려요 Dan!! 사실 인프라 작업을 위주로 진행을 하다가 PR을 위해(조금이라도 코드 리뷰를 받기위해) 부랴부랴 도메인 설계 작업을 진행하느라 부족했었던 부분이 많았을텐데 하나하나 얘기해주셔서 감사드렸습니다...!! 다음 PR에는 이번에 주신 리뷰들을 잘 반영하여 좀 더 나은 코드로 PR을 해보겠습니다! 💪

ikjo93 committed:
refactor : 1주차(수요일) 리뷰사항 반영

* DB 필드명 가독성 개선
 - reservation_date_time -> reservation_datetime
 - positionx -> position_x
 - preice_day -> priceperday

* Facility 클래스 필드 통합
 - airConditioner, hariDryer 등 부수적인 필드를 option(String) 필드로 통합
